src/platform/forms-system/src/js/review/ReviewCollapsibleChapter.jsx

```diff
@@ -146,6 +146,24 @@ export default class ReviewCollapsibleChapter extends React.Component {
             });
             const title = page.reviewTitle || page.title || '';
 
+            const hasVisibleFields =
+              pageSchema &&
+              Object.entries(pageSchema.properties)
+                .map(entry => entry[0])
+                .filter(propName =>
+                  showReviewField(
+                    propName,
+                    pageSchema,
+                    pageUiSchema,
+                    form.data,
+                    formContext,
+                  ),
+                ).length > 0;
+
+            if (pageSchema && !hasVisibleFields) {
+              return null;
+            }
```

Comment:
`hasVisibleFields` is a compound boolean expression that already includes `pageSchema`, so do we need to have `pageSchema && !hasVisibleFields` as a condition? If not, I think `if (!hasVisibleFields)` would be fine.